tests/crypto_test.py

```diff
@@ -274,8 +307,12 @@ def test_signed_only_false(self):
 
     @staticmethod
     def _context_mock():
-        error = gpgme.GpgmeError()
-        error.code = gpgme.ERR_AMBIGUOUS_NAME
+        # We need an uninittialized class that is a descendent of GPGMEError
+        # for the test, but we need to make the getcode method a mock, so we
+        # use type to make a dynamic sublcass, since raise wont' accept
+        # anything that isn't an Exception
+        error = type('CustomError', (gpg.errors.GPGMEError, ), {})
+        error.getcode = mock.Mock(return_value=gpg.errors.AMBIGUOUS_NAME)
```

Comment:
Is there any reason for this rather cryptic code? What would be wrong with a local class:
```py
class CustomError(gpg.errors.GPGMEError):
    getcode = mock.Mock(return_value=gpg.errors.AMBIGUOUS_NAME)
```
Or what do you mean by "uninitialzed" here?